### PR TITLE
Change the validation credentials of authorization

### LINKED ISSUE #123: Login occurs under guest if there is no password
Spring-data properties
```properties
tarantool.username=admin1
tarantool.password=
```

Error
```java
io.tarantool.driver.exceptions.TarantoolInternalException: InnerErrorMessage:
code: 42
message: Execute access to function 'repository.get' is denied for user 'guest'
```

### PR DESCRIPTION
**Pull request by ArtDu**
Now the connection to Tarantool under the guest user goes according
to the documentation rules: Username = "guest" or "", Password = "".
Previously, logging took place under guest, if you specify any name
without a password, for example:
Username = "notGuestName" or "", Password = ""

Closes #123

### CODE REVIEW

File: src/test/java/io/tarantool/driver/integration/ConnectionIT.java [hunk 1]
```diff
@@ -102,6 +105,67 @@ public void connectAndCloseAfterFuture() throws Exception {
...
+    public void testRandomNameUserWithoutPassword_shouldThrowException() throws Exception {
+        TarantoolCredentials credentials = new SimpleTarantoolCredentials(
+                "RandomUserName", "");
+        TarantoolServerAddress serverAddress = new TarantoolServerAddress(
+                tarantoolContainer.getHost(), tarantoolContainer.getPort());
+        try (ClusterTarantoolTupleClient client = new ClusterTarantoolTupleClient(credentials, serverAddress)) {
+            try {
+                List<?> resultList = client.eval("return box.session.user()").get();
+                String sessionUser = (String) resultList.get(0);
+                assertEquals(GUEST_USER, sessionUser);
```
Comment (vrogach2020) [inline src/test/java/io/tarantool/driver/integration/ConnectionIT.java:160]: This line is useless.
We must call fail() here if I understand the logic correctly.

Also to improve readability of test please try use assertThrows / assertDoesNotThrow check methods when possible.

**Review by vrogach2020 — CHANGES_REQUESTED** [review]

File: src/test/java/io/tarantool/driver/integration/ConnectionIT.java (hunk 1 above)
Comment (ArtDu) [reply src/test/java/io/tarantool/driver/integration/ConnectionIT.java]: Thank you for your comment, rewrote everything into assertThrows

File: src/main/java/io/tarantool/driver/auth/ChapSha1TarantoolAuthenticator.java [hunk 2]
```diff
@@ -36,7 +36,7 @@ public TarantoolAuthMechanism getMechanism() {
      */
     @Override
     public boolean canAuthenticateWith(SimpleTarantoolCredentials credentials) {
-        return !credentials.isEmpty();
+        return !credentials.isGuest();
```
[the remark this answers is not in the document]
Comment (ArtDu) [reply src/main/java/io/tarantool/driver/auth/ChapSha1TarantoolAuthenticator.java]: I did as you suggested, also added a separate exception TarantoolBadCredentialException

File: src/test/java/io/tarantool/driver/integration/ConnectionIT.java [hunk 3]
```diff
@@ -102,6 +105,67 @@ public void connectAndCloseAfterFuture() throws Exception {
...
+                tarantoolContainer.getHost(), tarantoolContainer.getPort());
+        try (ClusterTarantoolTupleClient client = new ClusterTarantoolTupleClient(credentials, serverAddress)) {
+            List<?> resultList = client.eval("return box.session.user()").get();
+            String sessionUser = (String) resultList.get(0);
+            assertEquals(GUEST_USER, sessionUser);
+        }
+    }
+
+    @Test
+    public void testExistingUserWithoutPassword() throws Exception {
```
[the remark this answers is not in the document]
Comment (ArtDu) [reply src/test/java/io/tarantool/driver/integration/ConnectionIT.java:175]: Added

File: src/test/java/io/tarantool/driver/integration/ConnectionIT.java [hunk 4]
```diff
@@ -102,6 +105,67 @@ public void connectAndCloseAfterFuture() throws Exception {
...
+                tarantoolContainer.getHost(), tarantoolContainer.getPort());
+        try (ClusterTarantoolTupleClient client = new ClusterTarantoolTupleClient(credentials, serverAddress)) {
+            List<?> resultList = client.eval("return box.session.user()").get();
+            String sessionUser = (String) resultList.get(0);
+            assertEquals(EXISTING_USER_WITHOUT_PASSWORD, sessionUser);
+        }
+    }
+
+    @Test
+    public void testRandomNameUserWithoutPassword_shouldThrowException() throws Exception {
```
[the remark this answers is not in the document]
Comment (ArtDu) [reply src/test/java/io/tarantool/driver/integration/ConnectionIT.java:189]: Added

**Review by vrogach2020 — APPROVED** [review]

**Pull request merged**